src/client/actions.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Actions are the mechanism through which notifications send messages back to the application that raised them. The
+ * service defines a number of ways in which actions can be raised (typically user interactions, such as clicking a
+ * button), and it is up to each application to decide if it wishes to be informed of these interactions.
+ *
+ * For an application to receive an event on each of these actions, the application should specify the message it
```

Comment:
* poke *

Yeah, could be worth a couple of minutes to think about how best to word some of this.

Reply:
How's this?

"For an application to be made aware when one of these actions occurs, it should listen for the {@link NotificationActionEvent|`notification-action`} event using the {@link addEventListener|`addEventListener`} function. This event will be raised once for each action that occurs on a notification created the application.

For each action, the application should specify the message it wishes to receive within the `NotificationActionEvent` when that action occurs. This is known as the {@link NotificationActionResult|action result}."

Note that I've made a call that events are "raised", but actions "occur". So second sentence in the first paragraph would become "The service defines a number of actions that can occur".

Reply:
There's a bit of an "if a tree falls in the woods..." scenario here. In my mind, if an app doesn't specify a result then there's no action. The wording here implies that there is an action, but the app wouldn't be informed about it.

If the user interacts with a notification from a non-running app, we use the presence of an action to decide if/when to re-launch that app. The "there is only an action if you specify a result" description makes more sense in that kind of context, compared to the "the action still happens, but since you weren't listening to it, we won't start-up your app.

Does that make sense?.. Still agree there's better wordings than what was there originally, but coming up with that wording is difficult. :-)